### PR TITLE
Allow setting a name prefix for agents

### PR DESCRIPTION
**Pull request by julienduchesne**
This will allows us to use `drone-{id}` instead of `agent-{id}`

### CODE REVIEW

File: config/load_test.go [hunk 1]
```diff
@@ -215,7 +215,8 @@ var jsonConfig = []byte(`{
     "Image": "drone/drone-runner-docker:latest",
     "Concurrency": 2,
     "KeepaliveTime": 360000000000,
-    "KeepaliveTimeout": 30000000000
+    "KeepaliveTimeout": 30000000000,
+	"NamePrefix": "agent-"
```
Comment (inkel) [inline config/load_test.go:219]: You might want to run `go fmt` on this file.

File: engine/planner.go [hunk 2]
```diff
@@ -108,9 +109,13 @@ func (p *planner) alloc(ctx context.Context, n int) error {
 	logger := logger.FromContext(ctx)
 	logger.Debugf("allocate %d servers", n)
 
+	namePrefix := p.namePrefix
+	if namePrefix == "" {
+		namePrefix = "agent-"
+	}
```
Comment (jvrplmlmn) [inline engine/planner.go:115]: Is this block necessary given that we default to `agent-`?

```go
NamePrefix string `envconfig:"DRONE_AGENT_NAME_PREFIX" default:"agent-"`
```

Alternatively, would it be also necessary to have this in `server/servers.go`?

Comment (julienduchesne) [reply engine/planner.go:115]: `server/servers.go` uses the config directly which has a default so it's not needed while the planner is a separate struct. It doesn't necessarily draw from the config. At least that's my iterpretation

Comment (inkel) [reply engine/planner.go:115]: Same interpretation here.

**Pull request merged**